integration/benchmark_test.go

```diff
@@ -0,0 +1,215 @@
+		expectedOS:    analyzer.OS{Name: "2 (Karoo)", Family: "amazon"},
+	},
+	{
+		name:          "happy path, debian:buster",
+		imageName:     "debian:buster",
+		imageFile:     "testdata/fixtures/debian-buster.tar.gz",
+		expectedFiles: []string{"var/lib/dpkg/status", "etc/debian_version", "etc/os-release", "usr/lib/os-release", "/config"},
+		expectedOS:    analyzer.OS{Name: "10.1", Family: "debian"},
+	},
+	{
+		name:          "happy path, photon:1.0",
+		imageName:     "photon:1.0-20190823",
+		imageFile:     "testdata/fixtures/photon-10.tar.gz",
+		expectedFiles: []string{"var/lib/rpm/Packages", "etc/lsb-release", "etc/os-release", "/config", "usr/lib/os-release"},
+		expectedOS:    analyzer.OS{Name: "1.0", Family: "photon"},
+	},
+	{
+		name:          "happy path, registry.redhat.io/ubi7",
+		imageName:     "registry.redhat.io/ubi7",
+		imageFile:     "testdata/fixtures/ubi-7.tar.gz",
+		expectedFiles: []string{"etc/redhat-release", "etc/system-release", "/config", "var/lib/rpm/Packages", "etc/os-release"},
+		expectedOS:    analyzer.OS{Name: "7.7", Family: "redhat"},
+	},
+	{
+		name:          "happy path, opensuse leap 15.1",
+		imageName:     "opensuse/leap:latest",
+		imageFile:     "testdata/fixtures/opensuse-leap-151.tar.gz",
+		expectedFiles: []string{"usr/lib/os-release", "usr/lib/sysimage/rpm/Packages", "/config", "etc/os-release"},
+		expectedOS:    analyzer.OS{Name: "15.1", Family: "opensuse.leap"},
+	},
+	{
+		name:                 "happy path, vulnimage with lock files",
+		imageName:            "knqyf263/vuln-image:1.2.3",
+		imageFile:            "testdata/fixtures/vulnimage.tar.gz",
+		expectedFiles:        []string{"etc/os-release", "node-app/package-lock.json", "python-app/Pipfile.lock", "ruby-app/Gemfile.lock", "rust-app/Cargo.lock", "/config", "etc/alpine-release", "lib/apk/db/installed", "php-app/composer.lock"},
+		expectedOS:           analyzer.OS{Name: "3.7.1", Family: "alpine"},
+		expectedLibraries:    "testdata/goldens/knqyf263vuln-image:1.2.3.expectedlibs.golden",
+		expectedPkgsFromCmds: "testdata/goldens/knqyf263vuln-image:1.2.3.expectedpkgsfromcmds.golden",
+	},
+}
+
+// benchCache is shared across benchmarks
+var benchCache string
+
+func runChecksBench(b *testing.B, ac analyzer.Config, ctx context.Context, tc testCase) {
+	for i := 0; i < b.N; i++ {
+		actualFiles, err := ac.Analyze(ctx, tc.imageFile)
+		require.NoError(b, err)
+
+		osFound, err := analyzer.GetOS(actualFiles)
+		require.NoError(b, err)
+
+		_, err = analyzer.GetPackages(actualFiles)
+		require.NoError(b, err)
+
+		_, err = analyzer.GetPackagesFromCommands(osFound, actualFiles)
+		require.NoError(b, err)
+
+		_, err = analyzer.GetLibraries(actualFiles)
+	}
+}
+
+func BenchmarkFanal_Library_DockerMode_WithoutCache(b *testing.B) {
+	benchCache, _ = ioutil.TempDir("", "TestFanal_Library_*")
+	for _, tc := range testCases {
+		tc := tc // save a copy of tc for use in t.Run https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721
```

Comment:
Even if we don't run benchmarks in parallel, is this line necessary?

Reply:
good catch - no we don't.